doc/whats_new/v1.4.rst

```diff
@@ -58,3 +58,6 @@ TODO: update at the time of the release.
   :meth:`base.OutlierMixin.fit_predict` now accept ``**kwargs`` which are
   passed to the ``fit`` method of the the estimator. :pr:`26506` by `Adrin
   Jalali`_.
+- |Enhancement| The default value in the `n_components` parameter of :func:`non_negative_factorization` in :class:`decomposition.NMF` and
+  :class:`decomposition.MiniBatchNMF` will change from `None` to `auto` in version 1.6, so that its value is automatically computed when H is provided.
+  :pr:`26634` by :user:`Alexandre Landeau <AlexL>` and :user:`Alexandre Vigny <avigny>`.
```

Comment:
```suggestion

:mod:`sklearn.decomposition`
............................

- |Enhancement| An "auto" option was added to the `n_components` parameter of
  :func:`decomposition.non_negative_factorization`, :class:`decomposition.NMF` and
  :class:`decomposition.MiniBatchNMF` to automatically infer the number of components
  when using a custom initialization. The default value of this parameter will change
  from `None` to `auto` in version 1.6.
  :pr:`26634` by :user:`Alexandre Landeau <AlexL>` and :user:`Alexandre Vigny <avigny>`.
```

Reply:
fixed in b3529bb1a0b276990b2124907d9b2f9b8f790b5d